chain/types/ethtypes/eth_types.go

```diff
@@ -799,6 +799,45 @@ func GetContractEthAddressFromCode(sender EthAddress, salt [32]byte, initcode []
 	return ethAddr, nil
 }
 
+// EthEstimateGasParams handles raw jsonrpc params for eth_estimateGas
+type EthEstimateGasParams struct {
+	Tx       EthCall
+	BlkParam *EthBlockNumberOrHash
```

Comment:
nit: it might be easier if we set this to "latest" by default rather than making it nullable.

Reply:
I actually prefer to handle that logic in the eth handler

Reply:
Fair enough.